camel/societies/role_playing.py

```diff
@@ -44,6 +46,8 @@ class RolePlaying:
             agent. (default: :obj:`False`)
         with_critic_in_the_loop (bool, optional): Whether to include a critic
             in the loop. (default: :obj:`False`)
+        critic_criteria (str, optional): Critic criteria for the critic agent.
```

Comment:
I think we should start reworking code into more dependency injection. This many parameters of the RolePlaying constructor point to low cohesion and breaking the single responsibility principle. Solution: Critic, TaskSpecifier, TaskPlanner, and arguably the two agents must be constructed outside RolePlaying and the instances provided to the RolePlaying constructor. The very refactoring that you are doing hints that sub-agents must be created outside and injected. May be out of scope of this PR.

Reply:
Agree. We can update that in the future PR.